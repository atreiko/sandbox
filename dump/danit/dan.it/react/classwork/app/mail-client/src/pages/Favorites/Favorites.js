import React from 'react'
import Email from '../../components/Email/Email';
import { emailsLoadingSelector, getEmailsSelector } from '../../store/emails/selectors';
import Loader from '../../components/Loader/Loader';
import { connect } from 'react-redux';

export function Favorites({ emails, isLoading }) {
  if (isLoading) {
    return <Loader />;
  }

  const emailCards = emails
    .filter(e => e.favorite)
    .map(e => <Email key={e.id} email={e} />);

  return (
    <div>
      {emailCards}
    </div>
  );
}

const mapStateToProps = (state) => {
  return {
    emails: getEmailsSelector(state),
    isLoading: emailsLoadingSelector(state)
  }
}

export default connect(mapStateToProps)(Favorites)
