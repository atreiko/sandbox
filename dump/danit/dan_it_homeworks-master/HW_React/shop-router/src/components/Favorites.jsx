import React, { useEffect, useState } from 'react';
import { Cards } from './Cards'

export const Favorites = () => {
    const [data, setData] = useState([])

    useEffect(() => {
        const itemFavorite = localStorage.getItem('Item');
        setData(JSON.parse(itemFavorite));
    },[])

    let cardsFavorite = data.map((item, index) => {
        return (
            <Cards
                name={item.name}
                id={item.id}
                price={item.price}
                article={item.article}
                photo={item.photo}
                url={item.url}
                key={index}
                item={item}
            />
        )
    })

return (
    <>
        <h1 className='title'>Favorite</h1>
        <div className='cards-container'>
            {cardsFavorite}
        </div>
    </>
)
}