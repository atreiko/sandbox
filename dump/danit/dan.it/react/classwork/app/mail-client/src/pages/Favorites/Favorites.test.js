import { render } from "@testing-library/react";
import { Favorites } from "./Favorites";

jest.mock('../../components/Email/Email', () => (props) => <div key={props.email.id} data-testid='test-email'>{props.email.topic}</div>)
jest.mock('../../components/Loader/Loader', () => () => <div data-testid='test-loader'>Loading...</div>)

const testEmails = [
  { id: 111, topic: 'Test email 111', body: 'Test body 111' },
  { id: 112, topic: 'Test email 112', body: 'Test body 112', favorite: true },
  { id: 113, topic: 'Test email 113', body: 'Test body 113', favorite: true }
]

describe('Testing Favorites.js', () => {
  test('Smoke test for Favorites', () => {
    render(<Favorites emails={testEmails} isLoading={false} />)
  })

  test('Favorites snapshot', () => {
    const { container } = render(<Favorites emails={testEmails} isLoading={false} />)
    expect(container.innerHTML).toMatchSnapshot()
  })

  test('Favorites page filters out non-favorite emails', () => {
    const { getAllByTestId } = render(<Favorites emails={testEmails} isLoading={false} />)

    const emailCards = getAllByTestId('test-email');
    expect(emailCards.length).toBe(2);
  })

  test('Loader is shown correctly', () => {
    const { getByTestId } = render(<Favorites isLoading={true} />)

    const loader = getByTestId('test-loader');
    expect(loader).toBeInTheDocument();
  })
})