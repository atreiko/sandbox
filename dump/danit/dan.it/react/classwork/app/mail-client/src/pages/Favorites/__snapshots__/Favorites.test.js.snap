// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Favorites.js Favorites snapshot 1`] = `"<div><div data-testid=\\"test-email\\">Test email 112</div><div data-testid=\\"test-email\\">Test email 113</div></div>"`;
